### PR TITLE
D3ASIM-2928: Moving simulation_calculation bases classes from d3a to …

### PR DESCRIPTION
**Pull request by faizan2590**
…d3a-interface.

### CODE REVIEW

File: d3a_interface/utils.py [hunk 1]
```diff
@@ -182,3 +182,44 @@ def get_area_name_uuid_mapping(serialized_scenario, mapping={}):
             new_mapping = get_area_name_uuid_mapping(child, mapping)
             mapping.update(new_mapping)
     return mapping
+
+
+def round_floats_for_ui(number):
+    return round(number, 3)
+
+
+def create_subdict_or_update(indict, key, subdict):
```
Comment (hannesdiedrich) [inline d3a_interface/utils.py:191]: Duplicate of `create_or_update_subdict` right?

**Review by hannesdiedrich — APPROVED** [review]
One comment, once addressed, LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

File: d3a_interface/constants_limits.py [hunk 2]
```diff
@@ -112,6 +112,7 @@ class LoadSettings:
         INITIAL_BUYING_RATE_LIMIT = RangeLimit(0, 10000)
         # Max load energy rate, in ct/kWh
         FINAL_BUYING_RATE_LIMIT = RangeLimit(0, 10000)
+        LOAD_PENALTY_RATE = 29.2
```
Comment (hannesdiedrich) [inline d3a_interface/constants_limits.py:115]: Please remove these from the d3a.constants. Thanks!

**Review by hannesdiedrich — APPROVED** [review]
One comment for the d3a. LGTM

File: d3a_interface/utils.py (hunk 1 above)
Comment (faizan2590) [reply d3a_interface/utils.py:191]: Not really, `create_subdict_or_update`creates an empty dict if key not found, whereas `create_or_update_subdict ` creates a dict with `subdict` if key is not found

File: d3a_interface/constants_limits.py (hunk 2 above)
Comment (faizan2590) [reply d3a_interface/constants_limits.py:115]: Replaced d3a's LOAD_PENALTY_RATE with d3a-interface's LOAD_PENALTY_RATE  but forgot to remove it from d3a.constants, will do while finalising related PR.

**Review by hannesdiedrich — APPROVED** [review]
LGTM

File: d3a_interface/sim_results/__init__.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+
+def is_pv_node_type(area):
+    return area['type'] in ["PVStrategy", "PVUserProfileStrategy", "PVPredefinedStrategy",
+                            "PVExternalStrategy", "PVUserProfileExternalStrategy",
+                            "PVPredefinedExternalStrategy"]
+
+
+def is_finite_power_plant_node_type(area):
+    return area['type'] == "FinitePowerPlant"
```
Comment (spyrostz) [inline d3a_interface/sim_results/__init__.py:40]: Please also add the Forecast External strategies in these lists.

**Review by spyrostz — APPROVED** [review]
One important comment, LGTM otherwise

**Pull request merged**